src/components/EditableName.jsx

```diff
@@ -1,26 +1,47 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
 import { Grid, IconButton, TextField } from "@mui/material";
 import { Check, Clear, Edit } from "@mui/icons-material";
 
-class EditableName extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      editing: false,
-      newName: "",
-      error: "",
-    };
-  }
+const EditableName = ({ name, setName, number }) => {
+  const [editing, setEditing] = useState(false);
+  const [newName, setNewName] = useState("");
+  const [error, setError] = useState("");
 
-  componentDidMount = () => {
-    const { name } = this.props;
-    this.setState({ newName: name });
+  useEffect(() => {
+    setNewName(name);
```

Comment:
Couldn't this just be used as the default on line 8?

Reply:
Looks like it-- sorry I should've caught that!

Reply:
Updated in [4c7681f](https://github.com/unchained-capital/caravan/pull/342/commits/4c7681fe2badd60264d2566255b8660f7e6c4151)